src/Engine/Waypoint/Waypoints.cpp

```diff
@@ -106,6 +106,11 @@ Waypoints::WaypointNameTree::Add(WaypointPtr wp)
   TCHAR normalized_name[wp->name.length() + 1];
   NormalizeSearchString(normalized_name, wp->name.c_str());
   RadixTree<WaypointPtr>::Add(normalized_name, std::move(wp));
+  if (!wp->shortname.empty()) {
```

Comment:
This will crash. See also https://github.com/XCSoar/XCSoar/pull/219#discussion_r475554551 (which you did not resolve).